streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1479,7 +1479,7 @@ private void closeTaskDirty(final Task task, final boolean removeFromTasksRegist
         try {
             // we call this function only to flush the case if necessary
             // before suspending and closing the topology
-            task.prepareCommit();
+            task.flush();
```

Comment:
The `SteamTask.prepareCommit()` set a boolean flag `hasPending TxCommit` if using EOS and the `commitNeeded` flag is `true` - is it OK to bypass that? I'm thinking so but I'd like to confirm.